corehq/motech/openmrs/models.py

```diff
@@ -74,3 +78,17 @@ class OpenmrsImporter(Document):
 
     def __str__(self):
         return self.server_url
+
+    def should_import_today(self):
```

Comment:
nit: i think this was better earlier. The logic if the import should happen does not look like something that should be on the model. Having it as a util method makes more sense to me.

```
def should_import_today(importer):
    pass
```

Reply:
Lol, this debate has been going on in my head for at least a decade. As you probably know there are 2 ORM patterns: Active Record (Django ORMs are an example) and Data Mapper (SQLAlchemy offers this (but can also be used as an Active Record)). In HQ we use both patterns. This change uses Active Record, where the model's behaviour is encapsulated in its own methods. With the Data Mapper pattern, model methods are limited to loading and saving, and models are passed to functions or the methods of separate business objects.

I agree with you -- I like the Data Mapper pattern because it applies the single responsibility principle -- it only deals with persistence, not business logic. ... But I think we use Document subclasses as Active Records a lot in HQ, and I thought that this change was more consistent with that.

As we migrate from Couch to SQL we are using `dbaccessor`s, which are effectively Data Mappers. I think that's great. I'll let the debate in my head rage on a little longer, and mull over reverting this. :upside_down_face:

Reply:
still in favour of reverting this..